File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/StateColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StateRow } from "./StateRow";

export interface StateColumns {
    Id: Column<StateRow>;
    Title: Column<StateRow>;
    ShortName: Column<StateRow>;
}

export class StateColumns extends ColumnsBase<StateRow> {
    static readonly columnsKey = 'Masters.State';
    static readonly Fields = fieldsProxy<StateColumns>();
}

File: GXpert/GXpert.Web/Modules/Masters/State/StateColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.State")]
[BasedOnRow(typeof(StateRow), CheckNames = true)]
public class StateColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string ShortName { get; set; }
}